homeassistant/helpers/storage.py

```diff
@@ -109,8 +111,12 @@ async def async_load(self) -> dict | list | None:
 
     async def _async_load(self):
         """Load the data and ensure the task is removed."""
+        if STORAGE_SEMAPHORE not in self.hass.data:
+            self.hass.data[STORAGE_SEMAPHORE] = asyncio.Semaphore(MAX_LOAD_CONCURRENTLY)
```

Comment:
Do we even need to store this in hass.data?
Can't this just replace line 23? `STORAGE_SEMAPHORE = asyncio.Semaphore(MAX_LOAD_CONCURRENTLY)`

and then `async with STORAGE_SEMAPHORE:`

Reply:
I think the semaphore needs to be created inside the event loop to get access to the current loop.